sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -83,20 +84,52 @@ public void setLevel(@Nullable SentryLevel level) {
   }
 
   /**
-   * Returns the Scope's transaction
+   * Returns the Scope's transaction name.
    *
    * @return the transaction
    */
-  public @Nullable String getTransaction() {
+  public @Nullable String getTransactionName() {
+    final Transaction tx = this.transaction;
+    return tx != null ? tx.getTransaction() : null;
+  }
+
+  /**
+   * Sets the Scope's transaction.
+   *
+   * @param transaction the transaction
+   */
+  public void setTransaction(@NotNull String transaction) {
+    final Transaction tx = this.transaction;
+    if (tx != null) {
+      tx.setName(transaction);
+    }
+  }
+
+  /**
+   * Returns current active Span or Transaction.
+   *
+   * @return current active Span or Transaction or null if transaction has not been set.
+   */
+  @SuppressWarnings("JdkObsolete")
+  public ISpan getSpan() {
```

Comment:
Could you please clarify the use case for this? seem rather "expensive" I'm trying to understand why we need this

Reply:
Integrations have to have a way to create child spans on existing spans. Since current transaction is hold in scope, I find that this is a natural place to add a `getSpan` method that returns either a transaction - if it has no active child spans, or the latest active span otherwise.

If there are active spans, we need to take the latest active one. I optimised this method to avoid reversing the list and just iterating on it from the end to the beginning. If we would like to avoid iteration completely, we would need to keep it in a `SortedSet` but then, we cannot use `Date` object to hold timestamps as it does not hold microseconds and if two spans are created in the same millisecond we lose the ability to figure out which is the latest one.

`Scope#getSpan` is called only when integrations start child spans, since realistically there are not that many spans attached to transactions I don't think he decrease performance in a significant way.

If you have suggestions how this can be changed I am open :-)

Reply:
In py, child spans are created via Hub.startSpan which accesses the current scope internally. So this *could* be made private if you insist on it

Reply:
Having `startSpan` on the `Hub` would make things simpler. I wasn't sure if it can be added since it's not in the specs.

Reply:
`startSpan` sounds better to me as well, you could make this method package private for a quick win.